staging/src/k8s.io/apimachinery/pkg/watch/mux.go

```diff
@@ -219,6 +219,8 @@ func (m *Broadcaster) Action(action EventType, obj runtime.Object) {
 // false if dropped.
 func (m *Broadcaster) ActionOrDrop(action EventType, obj runtime.Object) bool {
 	select {
+	case <-m.stopped:
```

Comment:
I think this will still race and it should be something like this to give priority to m.stopped
```
	select {
	case <-m.stopped:
		return
	default:
	}
       select {
	case m.incoming <- Event{action, obj}:
		return true
	default:
		return false
	}
```

Reply:
otherwise it chooses one at random https://go.dev/tour/concurrency/5

Reply:
Seems similar to the patter that is used here:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L1329-L1350

Reply:
> otherwise it chooses one at random https://go.dev/tour/concurrency/5 

Ah interesting  I didn't know this, I'll update following the above semantics and add the unit test @aojea Basically wrote for me below :)

Reply:
Done